avocado/core/decorators.py

```diff
@@ -73,14 +73,28 @@ def skip(message=None):
     """
     Decorator to skip a test.
     """
-    def decorator(function):
-        if not isinstance(function, type):
-            @wraps(function)
-            def wrapper(*args, **kwargs):  # pylint: disable=W0613
-                raise core_exceptions.TestSkipError(message)
-            function = wrapper
-        function.__skip_test_decorator__ = True
-        return function
+    def decorator(obj):
+        def method_decorator(function):
+            if not isinstance(function, type):
+                @wraps(function)
+                def wrapper(*args, **kwargs):  # pylint: disable=W0613
+                    raise core_exceptions.TestSkipError(message)
+                function = wrapper
+            function.__skip_test_decorator__ = True
+            return function
+
+        def class_decorator(cls):
+            for key in cls.__dict__:
+                value = getattr(cls, key)
+                if callable(getattr(cls, key)):
+                    wrapped = method_decorator(value)
+                    setattr(cls, key, wrapped)
+            return cls
+
+        if inspect.isclass(obj):
```

Comment:
Looking at _inspect_ source (https://github.com/python/cpython/blob/master/Lib/inspect.py#L72), the method _isclass_ does `isinstance(object, type)`. In this case, I don't see a reason to use _inspect_, in favor of `isinstance(obj, type)`.

Reply:
Ok, I will change that.